internal/api/handlers.go

```diff
@@ -50,17 +53,22 @@ func GetRecommendationSetList(c echo.Context) error {
 	allRecommendations := []map[string]interface{}{}
 
 	for _, recommendation := range recommendationSets {
-		recommendationData := make(map[string]interface{})
-		recommendationData["id"] = recommendation.ID
-		recommendationData["cluster_uuid"] = recommendation.Workload.Cluster.ClusterUUID
-		recommendationData["cluster_alias"] = recommendation.Workload.Cluster.ClusterAlias
-		recommendationData["project"] = recommendation.Workload.Namespace
-		recommendationData["workload_type"] = recommendation.Workload.WorkloadType
-		recommendationData["workload"] = recommendation.Workload.WorkloadName
-		recommendationData["container"] = recommendation.ContainerName
-		recommendationData["last_report"] = recommendation.Workload.Cluster.LastReportedAtStr
-		recommendationData["values"] = recommendation.Recommendations
-		allRecommendations = append(allRecommendations, recommendationData)
+		if is_user_authorized_for_resource(types.ResourceObject{
+			Cluster: recommendation.Workload.Cluster.ClusterUUID,
+			Project: recommendation.Workload.Namespace,
+		}, user_permissions) {
+			recommendationData := make(map[string]interface{})
+			recommendationData["id"] = recommendation.ID
+			recommendationData["cluster_uuid"] = recommendation.Workload.Cluster.ClusterUUID
+			recommendationData["cluster_alias"] = recommendation.Workload.Cluster.ClusterAlias
+			recommendationData["project"] = recommendation.Workload.Namespace
+			recommendationData["workload_type"] = recommendation.Workload.WorkloadType
+			recommendationData["workload"] = recommendation.Workload.WorkloadName
+			recommendationData["container"] = recommendation.ContainerName
+			recommendationData["last_report"] = recommendation.Workload.Cluster.LastReportedAtStr
+			recommendationData["values"] = recommendation.Recommendations
```

Comment:
Not now but will it be possible to have structure of recommendationData so that it would be useful to initialize structure objects?